test/api/tenants.test.ts

```diff
@@ -0,0 +1,175 @@
+import { Root } from '../../../../src/core/server/root';
+import { describe, expect, it, beforeAll, afterAll } from '@jest/globals';
+
+jest.setTimeout(50000);
+
+describe('/api/create_tenant', () => {
+  const credentials = `Basic ${Buffer.from(
+    `admin:${process.env.OPENSEARCH_INITIAL_ADMIN_PASSWORD}`
+  ).toString('base64')}`;
+  let root: Root;
+
+  beforeAll(async () => {
+    root = osdTestServer.createRootWithSettings(
+      {
+        opensearch: {
+          hosts: ['https://localhost:9200'],
+          ignoreVersionMismatch: true,
+          ssl: { verificationMode: 'none' },
+          username: 'kibanaserver',
+          password: 'kibanaserver',
+        },
+        opensearch_security: {
+          multitenancy: {
+            enabled: true,
+            tenants: {
+              preferred: ['Private', 'Global'],
+            },
+          },
+        },
+      },
+      { dev: true }
+    );
+
+    console.log('Starting OpenSearchDashboards server');
+    await root.setup();
+    await root.start();
+    console.log('Started OpenSearchDashboards server');
+  });
+
+  afterAll(async () => {
+    await root.shutdown();
+  });
+
+  it('Fails if no tenant is given', async () => {
+    const response = await osdTestServer.request
+      .post(root, '/api/create_tenant')
+      .set('Authorization', credentials);
+
+    expect(response.status).toEqual(400);
+  });
+
+  it('Fails if the user is unauthorized', async () => {
+    const response = await osdTestServer.request
+      .post(root, '/api/create_tenant')
+      .send({ tenant: 'new_tenant_name' });
+
+    expect(response.status).toEqual(401);
+  });
+
+  it('Creates tenant, roles and dashboards', async () => {
+    const response = await osdTestServer.request
+      .post(root, '/api/create_tenant')
+      .set('Authorization', credentials)
+      .send({ tenant: 'new_tenant_name' });
+
+    expect(response.status).toEqual(200);
+
+    expect(response.body.data).toMatchObject({
+      tenant: "'new_tenant_name' created.",
+      roles: {
+        response: [
+          "'bap_plugins_visibility' created.",
+          "'bap_new_tenant_name_privileged_user_role' created.",
+          "'bap_new_tenant_name_pseudonymize_role' created.",
+          "'bap_new_tenant_name_user_role' created.",
+          "'bap_new_tenant_name_mordred_role' created.",
+        ],
+      },
+      backendRoles: {
+        response: [
+          "'bap_new_tenant_name_user_role' created.",
+          "'bap_plugins_visibility' created.",
+          "'bap_new_tenant_name_pseudonymize_role' created.",
+          "'bap_new_tenant_name_privileged_user_role' created.",
+        ],
+      },
+      dashboards: expect.anything(),
+    });
+
+    const roles = response.body.data.roles.response;
+    expect(roles).not.toContain(
+      "'bap_new_tenant_name_anonymous_access_role' created"
+    );
+  });
+
+  it('Creates anonymous role', async () => {
+    // Delete 'bap_plugins_visibility' role from last test
```

Comment:
Are tests always run in the same order, one after another? Normally tests should be independent from other tests.

Reply:
I changed it to clean up the database after each test so they are independent from each other.